src/test/java/org/junitpioneer/jupiter/resource/DirTests.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2016-2022 the original author or authors.
+ *
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the Eclipse Public License v2.0 which
+ * accompanies this distribution and is available at
+ *
+ * http://www.eclipse.org/legal/epl-v20.html
+ */
+
+package org.junitpioneer.jupiter.resource;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junitpioneer.jupiter.resource.Assertions.assertCanAddAndReadTextFile;
+import static org.junitpioneer.jupiter.resource.Assertions.assertEmptyReadableWriteableTemporaryDirectory;
+import static org.junitpioneer.testkit.assertion.PioneerAssert.assertThat;
+
+import java.nio.file.Path;
+
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import org.junitpioneer.testkit.ExecutionResults;
+import org.junitpioneer.testkit.PioneerTestKit;
+
+class DirTests {
```

Comment:
New thought: To make this directory easier to navigate, we should prefix test class names with their "topic", i.e. "Resource" or "TemporaryDirectory" where that makes sense.

```suggestion
class TemporaryDirectoryDirTests {
```

Reply:
Sounds good to me! I'll accept the suggestion.